GuildedRoseLLC/Items/RemoteItemRepository.swift

```diff
@@ -5,28 +5,30 @@ public class RemoteItemRepository: ItemRepository {
     public init() {}
     
     public func getItems(onSuccess: @escaping ([Item]) -> Void) {
-        var itemList: [Item] = []
-        
         let url = URL(string: "https://floating-spire-59497.herokuapp.com/api/v1/items")
         guard let requestURL = url else { fatalError("The internet is broken")}
-        
         let request = URLRequest(url: requestURL)
         
         let task = URLSession.shared.dataTask(with: request) {data,response,error in
             if error != nil {
                 return
             } else {
                 if let data = data {
-                    let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [[String: Any]]
-                    if let json = json {
-                        itemList = json.map{
-                            Item(json: $0)
-                        }
-                        onSuccess(itemList)
-                    }
+                    let itemList = self.parse(json: data)
+                    onSuccess(itemList)
                 }
             }
         }
         task.resume()
     }
+    
+    public func parse(json: Data) -> [Item] {
```

Comment:
It seems a little strange to me that your `RemoteItemRepository` exposes a public `parse` function. It looks like you did this in order to test it, but we might want to rethink either what module this `parse` function is defined in, how we're going about testing it, or both.

Reply:
extracted to new class ItemParser